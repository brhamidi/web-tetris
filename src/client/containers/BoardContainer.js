import { connect } from 'react-redux'
import { shapeDown } from '../actions'
import Board from '../components/board'

const mapStateToProps = (state, ownProps) => ({
		player: ownProps.player,
		board: state.board,
		currentShape: state.currentShape
})

const mapDispatchToProps = dispatch => ({
		shapeDown: shape => dispatch(shapeDown(shape))
})

export default connect(
		mapStateToProps,
		mapDispatchToProps
)(Board)
