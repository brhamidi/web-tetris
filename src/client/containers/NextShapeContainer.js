import { connect } from 'react-redux'
import NextShape from '../components/nextShape'

const mapStateToProps = (state) => ({
		shape: state.nextShape
})

export default connect(mapStateToProps)(NextShape)
