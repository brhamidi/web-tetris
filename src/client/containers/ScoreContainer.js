import { connect } from 'react-redux'
import Score from '../components/score'

const mapStateToProps = (state) => ({
	score: state.score,
	name: state.name
})

export default connect(mapStateToProps)(Score)
