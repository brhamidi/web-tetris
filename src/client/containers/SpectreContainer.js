import { connect } from 'react-redux'
import Spectre from '../components/spectre'

const mapStateToProps = (state) => ({
	mode: state.mode,
	spectre: state.spectre
})

export default connect(mapStateToProps)(Spectre)
