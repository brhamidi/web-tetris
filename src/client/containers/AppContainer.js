import React from 'react';

import ScoreContainer from '../containers/ScoreContainer';
import SpectreContainer from '../containers/SpectreContainer';
import NextShapeContainer from './NextShapeContainer';
import BoardContainer from './BoardContainer';

import { setStatusGame, GameStatus, setInfo} from '../actions';

import { connect } from 'react-redux';

import { List } from 'immutable';
import * as Styles from './app.css';

import io from "socket.io-client";

const mapStateToProps = state => {
	return {
		info: state.info,
		status: state.status
	}
}

const socket = io.connect("http://localhost:3000");

class App extends React.Component {
	constructor(props) {
		super(props);
	}

	componentDidMount() {
		this.getInfo(this.props.dispatch)()
	}

	getInfo(dispatch) {
		const url = window.location.hash.slice(1);
		const split1 = url.split(']');
		if (split1[1] == '' && split1[2] == undefined) {
			const split2 = split1[0].split('[');
			if (split2[2] == undefined && split2[0] && split2[1]) {
				const room = split2[0];
				const player = split2[1];
				return () => dispatch(setInfo(socket, room, player));
			}
			return () => dispatch(setStatusGame(GameStatus.ERROR));
		}
		return () => dispatch(setStatusGame(GameStatus.ERROR));
	}

	render() {
		// TODO  a parametric component for error message
		if (this.props.status == GameStatus.ERROR)
			return <h2> Error .. </h2>;
		if (this.props.status == GameStatus.LOADING)
			return <h2> Loading .. </h2>;
		if (this.props.info == 'full')
			return <h2> Session Full </h2>;
		if (this.props.info == 'started')
			return <h2> Game still already </h2>;
		return (
			<div style={Styles.appStyle} >
				<div style={Styles.headerStyle} >
					<h3> tetris </h3>
				</div>
				<ScoreContainer />
				<NextShapeContainer />
				<BoardContainer
					player={this.props.info}
					socket = {socket} />
				<SpectreContainer socket = {socket} />
			</div>
		);
	}
}

export default connect(mapStateToProps)(App);
