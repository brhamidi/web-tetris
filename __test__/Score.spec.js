import React from 'react';
import { shallow } from 'enzyme';
import Score from '../src/client/components/score';

import AppContainer from '../src/client/containers/AppContainer';
import SpectreContainer from '../src/client/containers/SpectreContainer';
import BoardContainer from '../src/client/containers/BoardContainer';
import NextShapeContainer from '../src/client/containers/NextShapeContainer';
import ScoreContainer from '../src/client/containers/ScoreContainer';

describe('Score', () => {
	test('renders correctly', () => {
		const wrapper = shallow(<Score />);
		expect(wrapper).toMatchSnapshot();
	});
});
