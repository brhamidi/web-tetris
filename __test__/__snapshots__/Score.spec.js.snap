// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "grey",
      "height": "100px",
      "width": "100%",
    }
  }
>
  <p>
     Score: 
  </p>
</div>
`;
